### PR TITLE
Expose pulumi log level via runner parameters

### PR DESCRIPTION
**Pull request by clarkb7**
### What does this PR do?

Adds runner parameters to control [pulumi logging](https://www.pulumi.com/docs/support/troubleshooting/#verbose-logging) to make debugging pulumi issues easier.

example `.test_infra_config.yaml` options
```yaml
configParams:
  pulumi:
    logLevel: 9
    logToStdErr: true
```

Like other options, can be set via env vars
- `E2E_PULUMI_LOG_LEVEL=9`
- `E2E_PULUMI_LOG_TO_STDERR=true`

### Motivation

pulumi's default log output doesn't contain enough information to debug issues

### Additional Notes

test-infra config model PR: https://github.com/DataDog/test-infra-definitions/pull/561

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 1]
```diff
@@ -39,6 +40,11 @@ const (
 var (
 	stackManager     *StackManager
 	initStackManager sync.Once
+
+	// Sets the log level for Pulumi operations
+	// Be careful setting this value, as it can expose sensitive information in the logs.
+	// https://www.pulumi.com/docs/support/troubleshooting/#verbose-logging
+	pulumiLogLevel = flag.Int("pulumi-log-level", 1, "Pulumi log level")
```
Comment (vboulineau) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:47]: I would not use a flag. We have the Param store that gets value for this purpose (taking from env vars)

Comment (pducolin) [reply test/new-e2e/pkg/utils/infra/stack_manager.go:47]: What about allowing using flags as a fallback for env variables in store ?

Comment (vboulineau) [reply test/new-e2e/pkg/utils/infra/stack_manager.go:47]: IMHO it does not really make sense, we already have flags in the `invoke` command. I don't think adding flags to `go test` is user friendly.

Comment (clarkb7) [reply test/new-e2e/pkg/utils/infra/stack_manager.go]: updated to use runner params instead. also added a `logToStdErr` option.

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 2]
```diff
@@ -241,7 +241,15 @@ func (sm *StackManager) getStack(ctx context.Context, name string, config runner
 		return nil, auto.UpResult{}, err
 	}
 
-	var loglevel uint = 1
+	logLevel, err := profile.ParamStore().GetIntWithDefault(parameters.PulumiLogLevel, 1)
```
[the remark this answers is not in the document]
Comment (clarkb7) [reply test/new-e2e/pkg/utils/infra/stack_manager.go]: added. thanks

**Review by vboulineau — APPROVED** [review]

**Pull request merged**